adr/ADR-8.md

```diff
@@ -122,8 +122,9 @@ type RoKV interface {
 	// History retrieves historic values for a key
 	History(ctx context.Context, key string) ([]Entry, error)
 
-	// WatchBucket watches the entire bucket for changes, all keys and values will be traversed including all historic values
-	WatchBucket(ctx context.Context) (Watch, error)
+	// WatchBucket watches the entire bucket for changes, all keys and values will be traversed including all historic values.
```

Comment:
So if I watch a key I get the old values first? Seems like maybe opt-in to that but default is last value.

Reply:
yes, only if you watch the entire bucket.  If you did a specific key watch you only get the most recent value. We would need something like Subject Filter on `foo.*` with a starting position of every last message in every subject - awkward. 

I think the sweet spot for history isnt going to be like 100s of historic values, but 10 (I might cap it at 10), so should be fine.

Reply:
Starting position of Last, but LastBySubject similar to what we added for msg limits?

Reply:
If this existed I wouldnt complain :)